File: DuDoong-Api/src/main/java/band/gosrock/api/ticketItem/service/GetEventOptionsUseCase.java
package band.gosrock.api.ticketItem.service;


import band.gosrock.api.ticketItem.dto.response.GetEventOptionsResponse;
import band.gosrock.api.ticketItem.mapper.TicketOptionMapper;
import band.gosrock.common.annotation.UseCase;
import lombok.RequiredArgsConstructor;

@UseCase
@RequiredArgsConstructor
public class GetEventOptionsUseCase {

    private final TicketOptionMapper ticketOptionMapper;

    public GetEventOptionsResponse execute(Long eventId) {

        return ticketOptionMapper.toGetEventOptionResponse(eventId);
    }
}

File: DuDoong-Api/src/main/kotlin/band/gosrock/api/ticketItem/service/GetEventOptionsUseCase.kt
package band.gosrock.api.ticketItem.service

import band.gosrock.api.ticketItem.dto.response.GetEventOptionsResponse
import band.gosrock.api.ticketItem.mapper.TicketOptionMapper
import band.gosrock.common.annotation.UseCase

@UseCase
class GetEventOptionsUseCase(
    private val ticketOptionMapper: TicketOptionMapper,
) {

    fun execute(eventId: Long): GetEventOptionsResponse =
        ticketOptionMapper.toGetEventOptionResponse(eventId)
}
